README.md

```diff
@@ -16,18 +16,47 @@
 
 This is the source code of the website of Apache Ratis.
 
+## Hugo Installation
+
 To render it you need hugo static site generator (https://gohugo.io/getting-started/installing) which is available for the most popular platforms as a single binary.
 
-To render the final website use the following command:
+On OSX, this can be installed via HomeBrew: `brew install hugo`. For other operating system, please refer to the
+aforementioned Hugo documentation for installation steps.
+
+## Building
+
+To render the final website, use the provided `build.sh` script. This script will generate the website in the directory
+`public/` and also perform a license check on the source files (prior to commit).
 
 ```
-hugo -d /destination/dir
+hugo
 ```
 
-To develop the site use
+To iteratively develop the website, you can use the `serve` command to start a local webserver with your content changes
+rendered in realtime:
 
 ```
 hugo serve
 ```
 
-which starts an internal server where you can always check the final rendered version.
+## Publishing website changes
+
+Committers must ensure that the state of the `asf-site-source` and `asf-site` branches are in sync at all times.
+Committers must never make changes directly to the `asf-site` branch.
+
+Content pushed to the `asf-site` branch is automatically published to the public website:
+https://ratis.incubator.apache.org
+
+There is no automation to automatically keep these branches in sync, but a general guide is to do the following. Beware
+that these steps are destructive!
+
+```bash
+$ git checkout asf-site-source && //hack hack hack
```

Comment:
When I was working on RATIS-583, I found it very helpful to clone both asf-site and asf-site-source into two local directories.
```bash
$ cd asf-site-source
$ hugo -d ../asf-site
$ cd asf-site
$ git diff
```
Indeed, I have two terminals so that I don't have to change directories.  See what you think about it.

Reply:
Oh, nice. I didn't realize you could just generate right into the other checkout. That's a bit more simple.